web-api/tests/split_documents.rs

```diff
@@ -0,0 +1,502 @@
+// Copyright 2023 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::collections::HashSet;
+
+use anyhow::Error;
+use itertools::Itertools;
+use reqwest::{Client, StatusCode};
+use serde::Deserialize;
+use serde_json::{json, Map, Value};
+use url::Url;
+use xayn_integration_tests::{
+    send_assert,
+    send_assert_json,
+    test_two_apps,
+    with_dev_options,
+    UNCHANGED_CONFIG,
+};
+use xayn_web_api::{Ingestion, Personalization};
+
+#[derive(Debug, Deserialize)]
+struct PersonalizedDocumentData {
+    id: String,
+    snippet_id: SnippetId,
+    score: f32,
+    // included by test only dev option
+    #[serde(default)]
+    snippet: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+struct SnippetId {
+    document_id: String,
+    snippet_idx: u32,
+}
+
+#[derive(Debug, Deserialize)]
+struct SearchResponse {
+    documents: Vec<PersonalizedDocumentData>,
+}
+
+impl SearchResponse {
+    fn id_ranking(&self) -> Vec<(&str, u32)> {
+        self.documents
+            .iter()
+            .map(|document| {
+                (
+                    &*document.snippet_id.document_id,
+                    document.snippet_id.snippet_idx,
+                )
+            })
+            .collect_vec()
+    }
+
+    fn sanity_checks(&self) {
+        let ordered = self
+            .documents
+            .iter()
+            .zip(self.documents.iter().skip(1))
```

Comment:
:art: instead of zipping two iterators we could slide a tuple window over one iterator (with `itertools`)
```suggestion
            .tuple_windows()
```